pydarn/io/borealis/restructure_borealis.py

```diff
@@ -0,0 +1,774 @@
+from pathlib2 import Path
+from typing import Union, List
+
+from pydarn import borealis_exceptions, BorealisUtilities, \
+                   borealis_formats
+
+
+class BorealisRestructureUtilities():
+    """
+    Utility class containing static methods for restructuring Borealis files.
+
+    Restructured files result in faster reads and a more human-readable
+    format. 
+
+    The shared fields of the file are written as a single value in the file, 
+    and unshared fields are written as arrays where number of records is the 
+    first dimension.
+
+    Static Methods
+    --------------
+    find_max_sequences(data)
+        Find the max number of sequences between records in a site file, for 
+        restructuring to arrays.
+    find_max_beams(data)
+        Find the max number of beams between records in a site file, for 
+        restructuring to arrays.
+
+    Class Methods
+    -------------
+    bfiq_site_to_array(filename, data_dict)
+        Convert bfiq site data to array style dictionary.
+    antennas_iq_site_to_array(filename, data_dict)
+        Convert antennas_iq site data to array style dictionary.
+    rawacf_site_to_array(filename, data_dict)
+        Convert rawacf site data to array style dictionary.
+    bfiq_array_to_site(filename, data_dict)
+        Convert bfiq array data to site record dictionary.
+    antennas_iq_array_to_site(filename, data_dict)
+        Convert antennas_iq array data to site record dictionary.
+    rawacf_array_to_site(data_dict)
+        Convert rawacf array data to site record dictionary. 
+    """
+
+    @staticmethod
+    def find_max_sequences(data: OrderedDict) -> int:
+        """
+        Finds the maximum number of sequences between records in a Borealis
+        site style data file.
+
+        Parameters
+        ----------
+        data
+            Site formatted data from a Borealis file, organized as one 
+            record for each slice
+        
+        Returns
+        -------
+        max_seqs
+            The largest number of sequences found in one record from the 
+            file
+        """
+        max_seqs = 0
+        for k in data:
+            if max_seqs < data[k]["num_sequences"]:
+                max_seqs = data[k]["num_sequences"]
+        return max_seqs
+
+    @staticmethod
+    def find_max_beams(data: OrderedDict) -> int:
+        """
+        Finds the maximum number of beams between records in a Borealis
+        site style data file.
+
+        Parameters
+        ----------
+        data
+            Site formatted data from a Borealis file, organized as one 
+            record for each slice
+        
+        Returns
+        -------
+        max_beams
+            The largest number of beams found in one record from the 
+            file
+        """
+        max_beams = 0
+        for k in data:
+
+            if max_beams < len(data[k]["beam_nums"]):
+                max_beams = len(data[k]["beam_nums"])
+        return max_beams
```

Comment:
Actually, they are not; they are slightly different in the if statements. If there is a way to generalize it, then awesome :+1:  Otherwise, I wouldn't worry too much about it and add a `TODO:` to refactor this code. 

Suggesting a generalization: if both return values are needed in the same places, you could calculate them in 1 for loop under one function/method and then return a tuple. 
Even if you don't need one of the return values you can use "-", function can look like this:
```python
for k in data:
    num_beams = len(data[k]["beam_nums"])
    num_sequences = data[k]["num_sequences"]

    if max_beams < num_beams:
             max_beams = num_beams
    if max_sequences < num_sequences:
             max_sequences = num_sequences 

return max_beams, max_sequences 
```

Reply:
some types do not have a 'beam_nums' key but do have 'num_sequences'